decidim-core/lib/decidim/coauthorable.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This concern contains the logic related to collective or shared authorship.
+  #
+  # Coauthorable shares nearly the same object interface as Authorable but with some differences.
+  # - `authored_by?(user)` is exactly the same
+  # - `normalized_authors` is pluralized
+  #
+  # All coauthors, including the initial author, will share the same permissions.
+  module Coauthorable
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :coauthorships, as: :coauthorable, class_name: "Decidim::Coauthorship", dependent: :destroy
+      has_many :authors, through: :coauthorships, class_name: "Decidim::User"
+      has_many :user_groups, as: :coauthorable, class_name: "Decidim::UserGroup", through: :coauthorships
+
+      scope :from_author, ->(author) { joins(:coauthorships).where("decidim_coauthorships.decidim_author_id": author.id) }
+      scope :from_user_group, ->(user_group) { joins(:coauthorships).where("decidim_coauthorships.decidim_user_group_id": user_group.id) }
+
+      # Checks whether the user is author of the given proposal, either directly
+      # authoring it or via a user group.
+      #
+      # user - the user to check for authorship
+      def authored_by?(user)
+        coauthorships.where(author: user).or(coauthorships.where(user_group: user.user_groups)).exists?
+      end
+
+      # Returns the identities for the authors, whether they are user groups or users.
+      #
+      # Returns an Array of User and/or UserGroups.
+      def identities
+        coauthorships.order(:created_at).collect(&:identity)
+      end
+
+      # Syntactic sugar to access first coauthor
+      def creator_author
+        authors.first
+      end
+
+      # Syntactic sugar to access first coauthor
+      def creator_user_group
+        user_groups.first
+      end
+
+      # Syntactic sugar to access first identity whether it is a User or a UserGroup.
+      def creator_identity
+        coauthorships.order(:created_at).first&.identity
+      end
+
+      # Adds a new coauthor to the list of coauthors. The coauthorship is created with
+      # current object as coauthorable and `user` param as author. To set the user group
+      # use `extra_attrs` either with `user_group` or `decidim_user_group_id` keys.
+      #
+      # @param user: The new coauthor.
+      # @extra_attrs: Extra
+      def add_coauthor(user, extra_attrs = {})
```

Comment:
This logic should probably belong to a command

Reply:
This method is used inside the corresponding create and update commands and in seeds I think this is its right place.

Reply:
Having this method here is:
- a way so that Create and Update commands can reuse code, both have to add coauthors.
- syntactic sugar so that commands and seeds can create objects in one line. This method name has more meaning than `Decidim::Coauthorship.create!(user_group: user_group)`
- add_coauthor is a task already tackled by these create and update commands, why add a sub-command empty of logic if is not needed?

By the way, I see that there are already other interfaces with this kind of features like [Resourceable#link_resources](https://github.com/decidim/decidim/blob/58a1ba7c3a16dfc3643ca8e459eca3497ffb83e8/decidim-core/lib/decidim/resourceable.rb#L61) so what I'm doing here is nothing new in the application.

Reply:
Ok! 👍